testing/test_text_layout.c

```diff
@@ -0,0 +1,116 @@
+#include "game/gui/text/text_layout.h"
+#include "resources/fonts.h"
+#include "utils/allocator.h"
+#include "utils/str.h"
+#include "video/surface.h"
+#include <CUnit/CUnit.h>
+
+void create_fake_font(font *font) {
+    font_create(font);
+    font->w = 8;
+    font->h = 8;
+    font->size = FONT_BIG;
+
+    // Characters up to and including ' ' and 'A' are 8x8
+    surface s;
+    for(int i = 0; i <= 33; i++) {
```

Comment:
This magic number 33 could use some explanation.
```c
// Create 8x8 surfaces for characters from ' ' (32) to 'A' (65), inclusive.
// We don't need to set surfaces for characters '\0' (0) through '\US' (31) because
// they are control characters, which the text renderer does not display.
for(int i = 0; i <= 'A' - ' '; i++) {
```